70_FLIPFluids/app_resources/render_common.hlsl

```diff
@@ -1,25 +1,25 @@
 #ifndef _FLIP_EXAMPLE_RENDER_COMMON_HLSL
 #define _FLIP_EXAMPLE_RENDER_COMMON_HLSL
 
-#ifdef __HLSL_VERSION
-struct VertexInfo
+struct SParticleRenderParams
 {
-    float4 position;
-	float4 vsSpherePos;
-    
     float radius;
+    float zNear;
+    float zFar;
     float pad;
-
-    float4 color;
-    float2 uv;
 };
 
-// obsolete
-struct GSInput
+struct VertexInfo
 {
-	float4 particle : TEXCOORD0;
+    float32_t4 position;
+	float32_t4 vsSpherePos;
+    float32_t radius;
+
+    float32_t4 color;
```

Comment:
why are you using 4D vectors for 3D quantities?

Reply:
why the duplication of `VertexInfo` and `PSInput` ?

Reply:
these weird vertices shouldn't exist because the pipeline that makes them shouldn't exist